api/src/main/java/org/apache/iceberg/types/GetProjectedIds.java

```diff
@@ -39,7 +39,7 @@
 
   @Override
   public Set<Integer> field(Types.NestedField field, Set<Integer> fieldResult) {
-    if (fieldResult == null) {
+    if (fieldResult == null || isEmptyStruct(field)) {
```

Comment:
The error here previously,  is that an empty struct has no fields, since it has no fields we never would ad an ID representing it (normally this happens by referring to subfields)

Reply:
Removed this change leaving the API behavior exactly the same, now instead we add on empty projections when doing the projection itself.